src/stdatamodels/validate.py

```diff
@@ -37,15 +37,9 @@ def _check_value(value, schema, ctx):
     """
     Perform the actual validation.
     """
-    if value is None:
-        if schema.get('fits_required'):
-            name = schema.get("fits_keyword") or schema.get("fits_hdu")
-            raise jsonschema.ValidationError("%s is a required value"
-                                              % name)
```

Comment:
I don't see `fits_required` in any of the jwst schemas:

```
grep -r fits_required jwst/datamodels/schemas | wc -l
```
nor in the MIRI schemas:
```
grep -r fits_required miri/datamodels/schemas | wc -l
```

Doesn't that mean it's an unused feature?  Or is there another batch of schemas I should be inspecting?

Reply:
Interesting.  I suspect workflow went like

1. Add `fits_required` validation
2. Add `fits_required` to some schemas
3. Realize this wasn't working how we expected it to and try bugfix.
4. Not be able to fix and/or give up.
5. Remove `fits_required` from schemas
6. Never remove the buggy code.

Issue spacetelescope/jwst#1083 discusses this in a bit of detail, but mostly the ship has sailed at that point.

Issue spacetelescope/jwst#2688 also gives some background insight into the use case.

It looks like it never worked as envisioned, and we never used it.

Reply:
I guess if we really wanted to have a required property, it would make sense to use (and make work) the JSON `required` property.  Currently it gets thrown to the wind if included in a schema because we walk the tree of the data, not the tree of the schema.